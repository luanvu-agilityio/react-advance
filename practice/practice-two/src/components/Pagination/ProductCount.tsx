import { ProductCountDisplay } from './PaginationStyles'

interface ProductCountProps {
  totalProducts: number
}

export const ProductCount = ({ totalProducts }: ProductCountProps) => (
  <ProductCountDisplay>
    <span>{totalProducts}</span>
    <span>Products</span>
  </ProductCountDisplay>
)

ProductCount.displayName = 'ProductCount'
