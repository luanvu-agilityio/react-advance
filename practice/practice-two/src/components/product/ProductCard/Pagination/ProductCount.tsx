import type { ProductCountProps } from 'types/Pagination'
import { memo } from 'react'
import { ProductCountDisplay } from './PaginationStyles'

export const ProductCount = memo(({ totalProducts }: ProductCountProps) => (
  <ProductCountDisplay>
    {totalProducts} Product{totalProducts !== 1 ? 's' : ''}
  </ProductCountDisplay>
))

ProductCount.displayName = 'ProductCount'
