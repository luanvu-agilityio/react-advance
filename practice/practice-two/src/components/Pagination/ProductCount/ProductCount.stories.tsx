import type { Meta, StoryObj } from '@storybook/react'
import { ProductCount } from './ProductCount'

const meta: Meta<typeof ProductCount> = {
  title: 'Components/Pagination/ProductCount',
  component: ProductCount,
  parameters: {
    layout: 'centered',
  },
  tags: ['autodocs'],
  argTypes: {
    totalProducts: {
      control: 'number',
      description: 'Total number of products to display',
    },
  },
}

export default meta
type Story = StoryObj<typeof ProductCount>

export const Default: Story = {
  args: {
    totalProducts: 42,
  },
}

export const SingleProduct: Story = {
  args: {
    totalProducts: 1,
  },
}

export const NoProducts: Story = {
  args: {
    totalProducts: 0,
  },
}

export const ManyProducts: Story = {
  args: {
    totalProducts: 1000,
  },
}
