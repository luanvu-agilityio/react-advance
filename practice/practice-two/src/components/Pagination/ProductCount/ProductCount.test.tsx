import { render } from '@testing-library/react'
import { ProductCount } from './ProductCount'

describe('ProductCount', () => {
  // SNAPSHOT TESTS FOR RENDERING
  describe('Rendering', () => {
    it('renders correctly for different product counts', () => {
      // Standard product count
      const { container: standardCount } = render(
        <ProductCount totalProducts={42} />
      )
      expect(standardCount).toMatchSnapshot('standard-count')

      // Zero products
      const { container: zeroCount } = render(
        <ProductCount totalProducts={0} />
      )
      expect(zeroCount).toMatchSnapshot('zero-count')

      // Single product (singular case)
      const { container: singleCount } = render(
        <ProductCount totalProducts={1} />
      )
      expect(singleCount).toMatchSnapshot('single-count')

      // Large number of products
      const { container: largeCount } = render(
        <ProductCount totalProducts={1000} />
      )
      expect(largeCount).toMatchSnapshot('large-count')
    })
  })
})
