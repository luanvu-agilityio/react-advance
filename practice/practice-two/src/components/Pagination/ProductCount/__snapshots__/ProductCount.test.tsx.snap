// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCount Rendering renders correctly for different product counts: large-count 1`] = `
<div>
  <span
    class="sc-kpOvIu bGasPY"
  >
    <span>
      1000
    </span>
    <span>
      Products
    </span>
  </span>
</div>
`;

exports[`ProductCount Rendering renders correctly for different product counts: single-count 1`] = `
<div>
  <span
    class="sc-kpOvIu bGasPY"
  >
    <span>
      1
    </span>
    <span>
      Products
    </span>
  </span>
</div>
`;

exports[`ProductCount Rendering renders correctly for different product counts: standard-count 1`] = `
<div>
  <span
    class="sc-kpOvIu bGasPY"
  >
    <span>
      42
    </span>
    <span>
      Products
    </span>
  </span>
</div>
`;

exports[`ProductCount Rendering renders correctly for different product counts: zero-count 1`] = `
<div>
  <span
    class="sc-kpOvIu bGasPY"
  >
    <span>
      0
    </span>
    <span>
      Products
    </span>
  </span>
</div>
`;
